llpc/context/llpcCompiler.cpp

```diff
@@ -556,28 +559,237 @@ Result Compiler::BuildShaderModule(const ShaderModuleBuildInfo *shaderInfo, Shad
 
   unsigned codeSize = ShaderModuleHelper::getCodeSize(shaderInfo);
   size_t allocSize = sizeof(ShaderModuleData) + codeSize;
-  unsigned *allocBuf =
-      static_cast<unsigned *>(shaderInfo->pfnOutputAlloc(shaderInfo->pInstance, shaderInfo->pUserData, allocSize));
+
+  ShaderModuleData moduleData = {};
+  unsigned *codeBufferArray = new unsigned[codeSize];
+  MutableArrayRef<unsigned> codeBuffer(codeBufferArray, codeSize / sizeof(unsigned));
+  memcpy(moduleData.hash, &hash, sizeof(hash));
+  Result result = ShaderModuleHelper::getModuleData(shaderInfo, codeBuffer, moduleData);
+
+  [[maybe_unused]] ResourcesNodes resourceNodes = {};
+  [[maybe_unused]] std::vector<ResourceNodeData> inputSymbolInfo;
+  [[maybe_unused]] std::vector<ResourceNodeData> outputSymbolInfo;
+  [[maybe_unused]] std::vector<ResourceNodeData> uniformBufferInfo;
+  [[maybe_unused]] std::vector<ResourceNodeData> storageBufferInfo;
+  [[maybe_unused]] std::vector<ResourceNodeData> textureSymbolInfo;
+  [[maybe_unused]] std::vector<ResourceNodeData> imageSymbolInfo;
+  [[maybe_unused]] std::vector<ResourceNodeData> atomicCounterSymbolInfo;
+  [[maybe_unused]] std::vector<ResourceNodeData> defaultUniformSymbolInfo;
```

Comment:
`[[maybe_unused]]` shouldn't apply to these. That attribute should be used for variables that may be unused depending on build options / preprocessor macros / etc. For example, variables that are only used by `assert`.